verification/src/solidity/compiler_fetcher.rs

```diff
@@ -0,0 +1,330 @@
+                    },
+                    DeserializedCompilerInfo {
+                        path: DownloadPath::Filename("solc-linux-amd64-v0.4.15+commit.8b45bddb".to_string()),
+                        long_version: ver("0.4.15+commit.8b45bddb"),
+                        sha256: H256::from_str("0xc71ac6c28bf3b1a425e77e97f5df67a80da3e4c047261875206561c0a110c0cb").unwrap(),
+                    },
+                    DeserializedCompilerInfo {
+                        path: DownloadPath::Filename("download/files/solc-linux-amd64-v0.4.16+commit.d7661dd9".to_string()),
+                        long_version: ver("0.4.16+commit.d7661dd9"),
+                        sha256: H256::from_str("0x78e0da6cad24ab145a8d17420c4f094c8314418ca23cff4b050bb2bfd36f3af2").unwrap(),
+                    },
+                    DeserializedCompilerInfo {
+                        path: DownloadPath::Filename("solc-linux-amd64-v10.8.9-nightly.2021.9.11+commit.e5eed63a".to_string()),
+                        long_version: ver("10.8.9-nightly.2021.9.11+commit.e5eed63a"),
+                        sha256: H256::from_str("0x791ee3a20adf6c5ab76cc889f13cca102f76eb0b7cf0da4a0b5b11dc46edf349").unwrap(),
+                    },
+                ]
+            })
+        ]);
+    }
+
+    fn assert_has_version(
+        releases: &HashMap<CompilerVersion, CompilerInfo>,
+        ver: &str,
+        expect: &str,
+    ) {
+        let ver = CompilerVersion::from_str(ver).unwrap();
+        let info = releases.get(&ver).unwrap();
+        let url = info.url.to_string();
+        assert_eq!(url, expect, "urls don't match");
+    }
+
+    #[test]
+    fn parse_releases() {
+        let list_json_file: ListJson = serde_json::from_str(DEFAULT_LIST_JSON).unwrap();
+        let download_url = Url::from_str(DEFAULT_DOWNLOAD_PREFIX).expect("valid url");
+        let releases = list_json_file.into_releases(&download_url).unwrap();
+        assert_has_version(
+            &releases,
+            "0.8.15-nightly.2022.5.27+commit.095cc647",
+            "https://github.com/blockscout/solc-bin/releases/download/solc-v0.8.15-nightly.2022.5.27%2Bcommit.095cc647/solc",
+        );
+        assert_has_version(
+            &releases,
+            "0.4.13+commit.0fb4cb1a",
+            "https://binaries.soliditylang.org/linux-amd64/solc-linux-amd64-v0.4.13+commit.0fb4cb1a",
+        );
+        assert_has_version(&releases,
+            "10.8.9-nightly.2021.9.11+commit.e5eed63a",
+            "https://binaries.soliditylang.org/linux-amd64/solc-linux-amd64-v10.8.9-nightly.2021.9.11+commit.e5eed63a"
+        );
+        assert_has_version(&releases,
+            "0.4.16+commit.d7661dd9",
+            "https://binaries.soliditylang.org/linux-amd64/download/files/solc-linux-amd64-v0.4.16+commit.d7661dd9"
+        );
+    }
+
+    #[tokio::test]
+    async fn list_download_releases() {
+        let config = Config::from_file(PathBuf::from_str("test_config.toml").unwrap()).unwrap();
```

Comment:
Nit: Wouldn't it be easier just to make default config and then change the `solidity.compilers_list_url` to required url in case if `target_os="macos"` (or windows?)?